### PR TITLE
Fix C89 build and silence warnings

### PR DESCRIPTION
**Pull request by hhromic**
## Description

The C89 build is currently broken because of a non-compliant comment. Fixed.
Also silenced the following simple warnings:

```
command.c:290:25: warning: format ‘%d’ expects argument of type ‘int *’, but argument 4 has type ‘unsigned int *’ [-Wformat=]
    if (sscanf(arg, "%x %d", &addr, &nbytes) != 2)
                        ~^          ~~~~~~~
                        %d
```

```
gfx/display_servers/dispserv_x11.c:76:13: warning: ‘x11_display_server_close_display’ defined but not used [-Wunused-function]
 static void x11_display_server_close_display(Display *dpy)
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gfx/display_servers/dispserv_x11.c:60:17: warning: ‘x11_display_server_open_display’ defined but not used [-Wunused-function]
 static Display* x11_display_server_open_display(void)
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
./libretro-common/include/libchdr/chd.h:231:38: warning: format ‘%d’ expects argument of type ‘int *’, but argument 3 has type ‘uint32_t *’ {aka ‘unsigned int *’} [-Wformat=]
 #define CDROM_TRACK_METADATA2_FORMAT "TRACK:%d TYPE:%s SUBTYPE:%s FRAMES:%d PREGAP:%d PGTYPE:%s PGSUB:%s POSTGAP:%d"
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(...and others like this...)
```

## Reviewers

@orbea

### CODE REVIEW

**Comment by orbea** [comment]
Mind sending the libchdr fixes upstream too?

https://github.com/rtissera/libchdr

**Comment by hhromic** [comment]
Sure, no problem. Is that actively maintained?

**Comment by orbea** [comment]
I think so, it doesn't seem to be active at the moment, but I have had changes from lbretro merged upstream before. I suspect the code bases could be merged more in other parts too, but I haven't gotten around to that yet...

Ideally it would be good to install libchdr as a system library, but that needs to be fixed upstream and I don't know cmake well enough.

**Comment by hhromic** [comment]
@orbea  I checked the upstream file now and the warning is actually triggered due to how the `libchd` library is used in `libretro-common/streams/chd_stream.c`, which is not part of the library.  So this fix is actually only relevant in the context of libretro-common.

**Comment by orbea** [comment]
I thought that upstream was supposed to be a generic library to be used in many instances, if it can be used correctly to trigger the warning I think it should be fixed upstream too?

**Comment by hhromic** [comment]
This particular warning is because the upstream-provided format defines expect signed ints (`%d`), for example in this define:

    #define GDROM_TRACK_METADATA_FORMAT       "TRACK:%u TYPE:%s SUBTYPE:%s FRAMES:%u PAD:%u PREGAP:%u PGTYPE:%s PGSUB:%s POSTGAP:%u"

These defines aren't used by the library itself but are there to be used by implementations such as RetroArch. However, libretro-common is using libchdr together with the following typedef:
```
typedef struct metadata {
   char type[64];
   char subtype[32];
   char pgtype[32];
   char pgsub[32];
   uint32_t frame_offset;
   uint32_t frames;
   uint32_t pad;
   uint32_t extra;
   uint32_t pregap;
   uint32_t postgap;
   uint32_t track;
} metadata_t;
```

Which uses unsigned integers (`uint32_t`) that are later used in `sscanf` calls with these format defines, triggering the warning. The libretro-common implementation is not wrong: tracks, pads, gaps are all unsigned integers after all. So neither upstream is incorrect nor RetroArch, but they must match.

In conclusion, if you aim to make libchdr a system-installed dependency, then libretro-common must change these numbers to be signed in the typedef if you don't want to see these warnings. Nothing to circumvent that unfortunately. But we can always keep the change locally like now :)
